iocore/net/NetVConnection.cc

```diff
@@ -56,7 +56,7 @@ NetVCOptions::get_proto_string() const
   default:
     break;
   }
-  return nullptr;
```

Comment:
The c++17 string_view allows for nullptr assignment.  The proper fix would be to fix ts::string_view.

Reply:
Do you have a reference? I checked and found nothing. What happens is the C-string constructor is invoked and it calls `strlen` on the `nullptr`, which crashes. As far as I can tell, that's by design.

Reply:
I tested this with gcc 7.2.1 on Fedora 26.

Reply:
I think the issue is that by the current definition..  char_traits::length() -> strlen() which by definition:
http://en.cppreference.com/w/cpp/string/byte/strlen 

does not accept nulls. Now if strlen() did take a null and return 0 then nullptr would work fine.

I think string_view() may yet get an update for this before to long.

Reply:
I checked the Boost and clang implementations - both crash in this case. I can see where gcc added a null check (not even an overload for `nullptr_t`). But such a change is, unfortunately, non-portable.